### PR TITLE
fix: explore dbt errors for all versions

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #6179 

### Description:

Screenshot 2023-07-12 at 14 20 18

### CODE REVIEW

File: packages/backend/src/dbt/dbtCliClient.ts [hunk 1]
```diff
@@ -140,15 +150,49 @@ export class DbtCliClient implements DbtClient {
...
                 );
+
+                if (!(e instanceof DbtError)) throw e;
+                errorLogs[dbtExec] = e.logs || [];
                 if (dbtExecs[dbtExecs.length - 1] === dbtExec) {
-                    throw e; // Throw last error
+                    throw new DbtError(
+                        `We failed to run "dbt ${command.join(
+                            ' ',
+                        )}" successfully across all version of dbt that Lightdash currently supports. You can see specific errors below for each dbt version we tried to use.`,
```
Comment (hamzahc1) [inline packages/backend/src/dbt/dbtCliClient.ts:167]: Typo - should read: "...across all version**s** of..."

Comment (hamzahc1) [inline packages/backend/src/dbt/dbtCliClient.ts:167]: Replace "dbt version we tried to use." with "dbt version we tried to compile with."

**Review by hamzahc1 — CHANGES_REQUESTED** [review]
One small typo and a copy change

File: packages/backend/src/dbt/dbtCliClient.ts (hunk 1 above)
Comment (ZeRego) [reply packages/backend/src/dbt/dbtCliClient.ts:167]: Is not always during the compile command. This error message is returned independently of the dbt command.

Comment (hamzahc1) [reply packages/backend/src/dbt/dbtCliClient.ts:167]: ok then leave as is!

**Review by rephus — APPROVED** [review]
That look great :+1:

File: packages/backend/src/dbt/dbtCliClient.ts (hunk 1 above)
Comment (ZeRego) [inline packages/backend/src/dbt/dbtCliClient.ts:167]:
```suggestion
                        )}" successfully across all versions of dbt that Lightdash currently supports. You can see specific errors below for each dbt version we tried to use.`,
```

**Pull request merged**